calico_cni/calico_cni.py

```diff
@@ -336,14 +357,18 @@ def _create_endpoint(self, assigned_ip):
                                                     self.container_id,
                                                     [assigned_ip])
         except AddrFormatError:
-            _log.error("This node is not configured for IPv%s", assigned_ip.version)
-            # TODO - call release_ip / cleanup
-            sys.exit(1)
+            self.env[CNI_COMMAND_ENV] = CNI_CMD_DELETE
```

Comment:
I don't think we should be modifying `self.env` here - we should make a copy first.

e.g
```
env = self.env.copy()
env[CNI_COMMAND_ENV = CNI_CMD_DELETE
self._release_ip(env)
...